src/backup_restore.cpp

```diff
@@ -128,10 +128,165 @@ void BackupRestore::backupAndRestoreIpzVpd(types::IPZVpdMap& io_srcVpdMap,
                                            const std::string& i_srcPath,
                                            const std::string& i_dstPath)
 {
-    (void)io_srcVpdMap;
-    (void)io_dstVpdMap;
-    (void)i_srcPath;
-    (void)i_dstPath;
+    if (!m_backupRestoreCfgJsonObj.contains("backupMap") ||
+        !m_backupRestoreCfgJsonObj["backupMap"].is_array())
+    {
+        logging::logMessage(
+            "Backup map not found in the backup and restore config JSON.");
+        return;
+    }
+
+    bool l_isEmptySrcVpdReceived = io_srcVpdMap.empty();
```

Comment:
Did not understand what are we trying to achieve here? Why can't we check these in place?

Reply:
I am not in agreement for the updating of the empty VPD Map

Reply:
removed l_isEmptySrcVpdReceived & l_isEmptyDstVpdReceived, updating io_srcVpdMap & io_dstVpdMap only if they are not empty.